### PR TITLE
DBZ-7291 Retry flush if failed with retriable exceptions each Dialect.

### PR DESCRIPTION
**Pull request by Naros**
https://issues.redhat.com/browse/DBZ-7291

Port of the PR from https://github.com/debezium/debezium-connector-jdbc/pull/55 after moving the repository.

### CODE REVIEW

File: documentation/modules/ROOT/pages/connectors/jdbc.adoc [hunk 1]
```diff
@@ -886,6 +886,15 @@ Fully-qualified names for fields are of the form `_fieldName_` or `_topicName_:_
...
 If you include this property in the configuration, do not set the `field.include.list` property.
 
+|[[jdbc-property-flush-max-retries]]<<jdbc-property-flush-max-retries, `+flush.max.retries+`>>
+|5
+|Specifies the maximum number of attempts to flush changes to the target database on specific database errors.
+If the number of retries is exhausted, the sink connector will enter a _FAILED_ state.
+
+|[[jdbc-property-flush-retry-delay-ms]]<<jdbc-property-flush-retry-delay-ms, `+flush.retry.delay.ms+`>>
+|1000
+|Specifies the number of milliseconds to wait in between flush attempts when a flush operation fails and is scheduled to be retried.
```
Comment (roldanbob) [inline documentation/modules/ROOT/pages/connectors/jdbc.adoc:896]:
```suggestion
|Specifies the number of milliseconds that the connector waits to retry a flush operation that failed.
```

Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/jdbc.adoc:896]: I think readers can infer that the value applies only if the operation is scheduled to be retried. If you disagree, we can restore that clause.

File: documentation/modules/ROOT/pages/connectors/jdbc.adoc [hunk 2]
```diff
@@ -886,6 +886,15 @@ Fully-qualified names for fields are of the form `_fieldName_` or `_topicName_:_
  +
 If you include this property in the configuration, do not set the `field.include.list` property.
 
+|[[jdbc-property-flush-max-retries]]<<jdbc-property-flush-max-retries, `+flush.max.retries+`>>
+|5
+|Specifies the maximum number of attempts to flush changes to the target database on specific database errors.
```
[the remark this answers is not in the document]
Comment (roldanbob) [reply documentation/modules/ROOT/pages/connectors/jdbc.adoc:891]: I'm not strongly attached to it, but I'm inclined to say "certain" vs. "specific" in the absence of information about exactly which errors might cause the failure.  To my mind, "certain" is more open-ended, whereas "specific" leads me to expect that we enumerate those errors somewhere.

**Review by roldanbob — CHANGES_REQUESTED** [review]
Hey @Naros I've proposed a few small changes. Let me know what you think. Thanks.

**Review by roldanbob — APPROVED** [review]
LGTM!

**Comment by jpechane** [comment]
@Naros Applied, thanks!

**Comment by mfvitale** [comment]
Since a combination of settings of these new retry configs can interfere with the `max.poll.interval.ms` causing a re-balance, should we add a note on the docs about it?

**Comment by jpechane** [comment]
@mfvitale Good idea

**Comment by mfvitale** [comment]
@Naros If you don't mind I'll can do it.

**Pull request merged**